File: LeetCode/Python3/LeetCode0933_NumberOfRecentCalls.py
# LeetCode0933_NumberOfRecentCalls.py
class RecentCounter:
	def __init__( self ):
		self.__q = collections.deque()

	def ping( self, t: int ) -> int:
		self.__q.append( t )

		th = t - 3000
		while ( th > self.__q[0]):
			self.__q.popleft()

		return len( self.__q )

File: LeetCode/C++/LeetCode0933_NumberOfRecentCalls.cpp
// LeetCode0933_NumberOfRecentCalls.cpp
// https://leetcode.com/problems/number-of-recent-calls/
// October LeetCoding Challenge Day 1
class RecentCounter {
public:
	queue<int> q;
	RecentCounter() {}

	int ping(int t) {
		q.push( t );
		while( !q.empty() && q.front() < t - 3000 )
			q.pop();
		return q.size();
	}
};